api/api.go

```diff
@@ -277,3 +276,42 @@ type IndexPostRequest struct {
 	InputFile  string          `json:"input_file"`
 	OutputFile string          `json:"output_file"`
 }
+
+type StorageKind string
+
+const (
+	UnknownStore StorageKind = ""
+	ArchiveStore StorageKind = "archivestore"
+	FileStore    StorageKind = "filestore"
+)
+
+func (k StorageKind) String() string {
+	return string(k)
+}
+
+func (k *StorageKind) Set(s string) error {
+	switch s := StorageKind(s); s {
+	case ArchiveStore, FileStore:
+		*k = s
+		return nil
+	}
+	return fmt.Errorf("unknown storage kind: %s", s)
+}
+
+type Config struct {
```

Comment:
Since this no longer imports as space.Config, I think it ought to be called StorageConfig.